drivers/vmwarevsphere/utils.go

```diff
@@ -35,6 +38,122 @@ func (d *Driver) remoteExec(procman *guest.ProcessManager, arg string) (int64, e
 	return code, nil
 }
 
+func (d *Driver) getDatastore(spec *types.VirtualMachineConfigSpec) (*object.Datastore, error) {
+	var sp *object.StoragePod
+	if d.DatastoreCluster != "" {
+		log.Infof("Finding datastore cluster %s", d.DatastoreCluster)
+		sp, err := d.finder.DatastoreCluster(d.getCtx(), d.DatastoreCluster)
+		if err != nil {
+			return nil, err
+		}
+
+		return d.recommendDatastore(sp, spec)
+	}
+
+	if d.Datastore != "" {
+		log.Infof("Finding datastore %s", d.Datastore)
```

Comment:
nit: shall it be "found"?

Reply:
the call underneath `return d.finder.Datastore(d.getCtx(), d.Datastore)` technically goes to a finder query and it accepts a string which was free form text before 2.3.3, technically it could not be found and fail. this is just a nicety for the UI since the next error you'd see would be "datastore xxx not found"

this also helps w/ defaulting as down below you will see i add that we're falling back so people know what we're doing if you pass in empty strings for datastore or datastore-clusters